forget: indicate why the oldest snapshot in a group is kept

konidev20 opened the pull request:
What does this PR change? What problem does it solve?
-----------------------------------------------------
When the oldest snapshot in the
list is retained, the reason is now prefixed with "oldest" to clearly
indicate why it's being kept. This change improves the clarity of retention
decisions, especially for edge cases involving the oldest available snapshot.

A snapshot is only marked as oldest if it's the last in the list AND its values matches the last seen value for that bucket

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #4806 

Linking the comment where the recommendation was made: 
https://github.com/restic/restic/issues/4806#issuecomment-2139842193

Checklist
---------
- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

konidev20 committed:
forget: indicate why the oldest snapshot in a group is kept
When the oldest snapshot in the
list is retained, the reason is now prefixed with "oldest" to clearly
indicate why it's being kept.

konidev20 committed:
forget: update tests to reflect specific reasons for keeping oldest snapshots in a group

MichaelEischer reviewed (changes requested):
Thanks for the PR. I think the "oldest" prefix should only be shown for snapshots that would be kept by the rule otherwise, see the comment for more details.

Please also extend the note in the documentation at https://github.com/restic/restic/blob/8a0edde40718630a6f31fd1bdb05c4af3dbb40b7/doc/060_forget.rst?plain=1#L217 to mention that these "oldest" snapshots are now marked with the "oldest " prefix.

konidev20 committed:
forget: make oldest snapshot marker more strict

Now, a snapshot is only marked as oldest if it's the last in the list AND its values matches the last seen value for that bucket.

Also, updated the corresponding golden files for the tests.

konidev20 committed:
docs: mention that the oldest snapshot is marked `oldest` in the reasons of the forget comman

konidev20 commented on internal/restic/snapshot_policy.go:
```diff
@@ -260,6 +260,9 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots, reason
 					if buckets[i].Count > 0 {
 						buckets[i].Count--
 					}
+					if nr == len(list)-1 {
```
Done. I have made the requested changes and also updated the tests.

MichaelEischer reviewed (approved):
LGTM. Thanks!
